tensorflow/lite/micro/kernels/reduce_test.cc

```diff
@@ -606,12 +549,351 @@ TF_LITE_MICRO_TEST(MeanInt84DWithoutKeepDimsWithPrecision) {
   int8_t expected_output_data_quant[2];
   int8_t input_data_quant[12];
 
-  tflite::testing::TestMeanOpQuantized<int8_t>(
+  tflite::testing::TestReduceOpQuantized<int8_t>(
       kInputShape4D, kInputData4D, input_data_quant, input_scale,
       input_zero_point, tflite::testing::kAxisShape4D,
       tflite::testing::kAxisData4D, kOutputShape4D, kGoldenData4D,
       output_data_quant, expected_output_data_quant, output_scale,
-      output_zero_point, &params);
+      output_zero_point, tflite::Register_MEAN(), &params, 1.0);
+}
+
+TF_LITE_MICRO_TEST(SumFloatFlatten2ReduceDims) {
```

Comment:
At first glance these tests don't seem to be directly ported from TFLite. Please let me know if I'm wrong and need to take a closer look. Otherwise, could you elaborate why TFLite tests were dropped and new ones were added? Please also elaborate how these expected outputs were obtained.

Reply:
This is actually due to github repo not being synced with google3, see b/236636177 for details.  I added the ported tests here in expectation of this being fixed before this PR got merged, but there's been no progress there.  Let me know how you'd like to proceed.

Reply:
The linked bug refers the 3 corresponding tests for Mean, not Sum. Is this the first time these 3 SumFlatten tests are added? If so, did you generate the golden data?

I'm still confused what the test porting strategy here is. For example, I see that we have `SumFloat(2/3/4)DKeepDims` here, but there are 12 `ConstFloatSumOpTest` tests in TFLite. Are those relevant? The other additions for Int8 and Int16 are fine.

Reply:
I'm not trying to port the tests from tensorflow/lite/kernels/reduce_test.cc, but rather add the equivalents of the mean op tests found in tensorflow/lite/micro/kernels/reduce_test.cc (this file).  I'm doing this in order to be consistent with the tests in this file, and thus my new tests should correspond 1:1 to the tests for the mean op in this file.  For example, mean has MeanFloat2DKeepDims so I'm adding SumFloat2DKeepDims, mean has MeanInt82DKeepDims so I'm adding SumInt82DKeepDims, etc and I'm reusing the same inputs as the mean op tests.  If you look at the mean op tests, they also don't correspond 1:1 to the tests from Lite.

Trying to keep the mean and sum op tests 1:1 *within this file* is also why I'm adding the three extra tests (SumFloatFlatten2ReduceDims, SumFloatFlatten2NonReduceDims, and SumFloatFlatten2MiddleDims) -- I expected that the Mean* version of these tests would be synced to the github repo so after that sync I'd be consistent.  This however never happened, and now I see that Advait actually rolled back these tests in cr/456546439 so perhaps it makes sense that I remove those tests as well?  Let me know what you prefer.

Also, if you prefer that I port the tests from lite directly rather than stay consistent with the tests within this file I'm fine with that, just let me know.  It's unclear to me what the guidelines here are.

Reply:
That makes sense. Let's keep all the tests that you've already created since you've already gone through the trouble of adding them. In addition, can we do another pass of looking at TFLite's `ConstFloatSumOpTest` and see what can be ported? That seems like the biggest gap to me right now. Irrelevant tests can be omitted. We should be good to go after this.

Reply:
Ported three more tests from lite.  The rest either don't make sense here or are already covered by existing tests, PTAL.

Reply:
Thanks for doing another pass.